pkg/controller/core/workload_controller.go

```diff
@@ -180,22 +186,18 @@ func (r *WorkloadReconciler) Reconcile(ctx context.Context, req ctrl.Request) (c
 	} else {
 		var updated, evicted bool
 		message := "The workload is deactivated"
+		dtCond := apimeta.FindStatusCondition(wl.Status.Conditions, kueue.WorkloadDeactivationTarget)
 		if !apimeta.IsStatusConditionTrue(wl.Status.Conditions, kueue.WorkloadEvicted) {
-			// The deactivation reason could be deduced as the maximum number of re-queuing retries if the workload met all criteria below:
-			//  1. The waitForPodsReady feature is enabled, which means that it has "PodsReady" condition.
-			//  2. The workload has already exceeded the PodsReadyTimeout.
-			//  3. The workload already has been re-queued previously, which means it doesn't have the requeueAt field.
-			//  4. The number of re-queued has already reached the waitForPodsReady.requeuingBackoffLimitCount.
-			if apimeta.IsStatusConditionFalse(wl.Status.Conditions, kueue.WorkloadPodsReady) &&
-				((wl.Status.RequeueState == nil && ptr.Equal(r.waitForPodsReady.requeuingBackoffLimitCount, ptr.To[int32](0))) ||
-					(wl.Status.RequeueState != nil && wl.Status.RequeueState.RequeueAt == nil &&
-						ptr.Equal(wl.Status.RequeueState.Count, r.waitForPodsReady.requeuingBackoffLimitCount))) {
-				message = fmt.Sprintf("%s due to exceeding the maximum number of re-queuing retries", message)
+			if dtCond != nil {
+				message = fmt.Sprintf("%s due to %s", message, dtCond.Message)
```

Comment:
we should also copy or concatenate the reason.

Reply:
Is it ok? 

```

Reply:
No, not in the message. It should be in the reason field, which is machine-friendly.

Reply:
Ah, got it. So in result we have for eviction condition `InactiveWorkloadRequeuingLimitExceeded` and for event `EvictedDueToInactiveWorkloadRequeuingLimitExceeded`.

Reply:
That event reason is long, but if the API validation allows it, go for it :)